docs/what-is-an-operator.md

```diff
@@ -1,2 +1,49 @@
 # What is an Operator?
 
+Operators are software extensions to Kubernetes that make use of custom resources to manage applications and their components. [Kubernetes documents](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)
+
+In the real world, we can define Operators: as a person who operates on some task(machine, equipment, process). Let's say it works on a given task "X". What qualities are required to peform the task X?
+
+We can classify an operator qualities into three categories:
+
+* Observe
+* Analyze
+* Act
+
+Observe the X work, analyze the state of X and finally act based on the current state of X to reach the desired state. Which will make the operators task completed, once we attain the desired state.
+
+Similar work/tasks are performed by an operators. It helps us to extend the Kubernetes API to allow custom functionality or behavior based on the needs of the consumer.
+
+# Operator Pattern 
+
+![Operator internals](images/operator-internals.png)
```

Comment:
What tool was used to create this png? Worried about maintainability.

Reply:
I created the diagram using google-slides. Either I can share the slide link or we can use something else which will be easier to maintain.